gtsam/hybrid/HybridFactorGraph.h

```diff
@@ -77,6 +43,11 @@ class HybridFactorGraph : protected FactorGraph<Factor>,
   using IsDC =
       typename std::enable_if<std::is_base_of<DCFactor, FACTOR>::value>::type;
 
+  /// Check if FACTOR type is derived from NonlinearFactor.
```

Comment:
why is that here? This class is not supposed to know about nonlinear factors

Reply:
Okay so should I move all the `IsNonlinear` based methods to `NonlinearHybridFactorGraph` and all the `IsGaussian` based methods to `GaussianHybridFactorGraph`? Then my previous comment on the benefits of templating come up since we can't move a lot of things to the .cpp file.